### PR TITLE
Page is an Entity

### PR DESCRIPTION
**Pull request by ad2476**
When building a Foundation doc, one way of representing pages might be to group them in with the rest of the Entities in the doc.

Downstream consumers of fnd.Entity can build up the appropriate document representations using Page's bbox to determine which entities are part of which page.

_Why not associate the entities on a page as its children_? It might be tricky to serialize such entity types -- protobuf messages aren't meant to be too huge, and a `repeated entities` field in the `Page` message could easily grow to thousands of entities large.

**Alternatively** `Page` could be a sort of metadata struct -- a foundation Document having its stream of entities as well as a list of `Page`s. `Page` could keep its data definition, but not explicitly as an `Entity`.

### CODE REVIEW

**Review by vontell — APPROVED** [review]

**Comment by vontell** [comment]
Debate/Idea: I actually think a page entity doesn't really need a special reason as to why it doesn't have children. Maybe the way to think about it is that the children are the entities that define the entity. In the case of an Address Entity, it is defined by the OCR words, the zip code, house number, etc... For all word-based entities, actually, it is defined by the words. But for a page, it is simply defined by an image region, or a region in a document, meaning that it does not have any children.

TLDR I want to make sure we don't have entities that are "special cases", and what this PR does makes sense, so I think we need to explicitly define why this makes sense.

**Review by andrey-ib — APPROVED** [review]
